app/models/order.rb

```diff
@@ -26,6 +26,7 @@ class Order < ApplicationRecord
 
   before_save :check_placed_on, if: %i[cancelled? status_changed?]
   after_commit :send_confirmation_email, if: :received?
+  after_update_commit :broadcast_order, if: :received?
```

Comment:
fix all the callbacks conditions as discussed on call

Reply:
Ok sir